mlrun/launcher/local.py

```diff
@@ -23,17 +40,291 @@ def verify_base_image(runtime):
     def save(runtime):
         pass
 
-    @staticmethod
-    def run(runtime):
-        pass
+    def run(
```

Comment:
I think BaseLauncher should implement `run` and the other launchers will implement `_run` so that we have common logic in the base launcher (like enrich and validate which can be extended by the child classes).
Also, wdyt about changing the method name to `launch`? I now think it may be more suitable and consistent with the class name.